### PR TITLE
Fixed occlusion algorithm and enhanced variable assignment

### PR DESCRIPTION
**Pull request by kazimierz-256**
Ola, Taras --- check --- this --- out --- the occlusion mechanism we've all been waiting for! NO CNF form breakdown, NO variable annihilation, NO NO NO. The basis for correctness of this computational idea lies in the following statement:
CONSEQ = (the cumulative consequence of all relevant causes postconditions) 
consider all valid assignments of fluents that satisfy CONSEQ where fluent F is true
if all of the valid assignments described above EXACTLY MATCH (excluding the fluent F itself) all valid assignments of fluents that satisfy CONSEQ where fluent F is FALSE then the fluent F shall be regarded as redundant and expelled from the garden of occlusion function.
Other fluents shall be kept in the occlusion function and rejoice.
Sample logic formulae where this idea works:
f and not f
(f or not f) and g
(f and g and h) or (not f and g and h) or (f and not g and h) or (not f and not g and h)

### CODE REVIEW

**Comment by kazimierz-256** [comment]
We need to fix unit tests to handle this new concept

File: apps/concurrent/test/or_story.plt [hunk 1]
```diff
@@ -4,13 +4,13 @@
 :- use_module("../modules/query_parsing.pl").
 :- use_module("../concurrent.pl").
 
-test(run_scenario) :-
-    parse_domain("A causes f or not f", Domain),
+possibly_executable :-
```
Comment (olabknbit) [inline apps/concurrent/test/or_story.plt]: why did you change that line? It makes the test not run.

Comment (kazimierz-256) [reply apps/concurrent/test/or_story.plt]: because I made a copypaste bug

**Review by kazimierz-256 — COMMENTED** [review]
All tests pass now

**Comment by kazimierz-256** [comment]
To remember:
no variable annihilates yet f shouldn't be in occlusion
g and (f or g)

**Comment by kazimierz-256** [comment]
Additionally, I have enhanced generation of future possible states (and initial state 0) so in optimistic case the computation time could be far less than exponential and I have deleted cumbersome code from previous brute-force approach. Now code that generates new states is much more elegant + it uses a nontrivial assignment algorithm. I guess this improvement is also much desired.

**Comment by kazimierz-256** [comment]
**TODO**: Requires changing the documentation by adding some brief explanation about the set of fluents that are excluded from modification
